shell/platform/android/image_external_texture_vk.cc

```diff
@@ -31,6 +32,50 @@ void ImageExternalTextureVK::Attach(PaintContext& context) {
 
 void ImageExternalTextureVK::Detach() {}
 
+sk_sp<flutter::DlImage> ImageExternalTextureVK::FindImage(uint64_t key) {
+  for (auto i = 0u; i < kImageReaderSwapchainSize; i++) {
+    if (images_[i].first == key) {
+      UpdateKey(key);
+      return images_[i].second;
+    }
+  }
+  return nullptr;
+}
+
+void ImageExternalTextureVK::UpdateKey(uint64_t key) {
+  if (keys_[0] == key) {
+    return;
+  }
+  auto i = 1u;
+  for (; i < kImageReaderSwapchainSize; i++) {
+    if (keys_[i] == key) {
+      break;
+    }
+  }
+  for (auto j = i; j > 0; j--) {
+    keys_[j] = keys_[j - 1];
+  }
+  keys_[0] = key;
+}
+
+void ImageExternalTextureVK::AddImage(sk_sp<flutter::DlImage> image,
+                                      uint64_t key) {
+  uint64_t lru_key = keys_[2];
```

Comment:
the [2] should be [kImageReaderSwapchainSize - 1]?

Reply:
Done